src/int_x64_backend.ml

```diff
@@ -7,24 +7,36 @@ let neg x = (-x)
 let add a b = a + b
 let sub a b = a - b
 let mul a b = a * b
-let div a b = a / b
+
+let unsigned_compare n m =
+  let open Nativeint in
+  compare (sub n min_int) (sub m min_int)
+
+let unsigned_div n d =
+  let open Nativeint in
+  if d < zero then
+    if unsigned_compare n d < 0 then zero else one
+  else
+    let q = shift_left (div (shift_right_logical n 1) d) 1 in
+    let r = sub n (mul q d) in
+    if unsigned_compare r d >= 0 then succ q else q
+
+let div a b = Nativeint.(to_int (unsigned_div (of_int a) (of_int b)))
 let rem a b = a mod b
 let succ x = x + 1
 let pred x = x - 1
-let abs x = Pervasives.abs x
+let abs x = if x >= 0 then x else (-x)
```

Comment:
```suggestion
let abs x = if x >= 0 then x else (-x)
let abs x =
  let mask = x asr Sys.int_size in (* extract sign: -1 if signed, 0 if not signed *)
  (x + mask) lxor mask
```
Branch-free version using [this bithacking trick](https://graphics.stanford.edu/~seander/bithacks.html#IntegerAbs).
Both implementations (current + my suggested) are the identity function when `x = min_int`.
Do you think that is better than making it be `0` or `max_int` in that case ?

Assembly for branch-free version (suggested) - I also checked the other variant they have (with xor first, then minus; that resulted in one extra instruction):
```asm
        movq    %rax, %rbx
        sarq    $62, %rbx
        orq     $1, %rbx
        leaq    -1(%rax,%rbx), %rax
        xorq    %rbx, %rax
        orq     $1, %rax
        ret
```

Asm for branching version (current):
```asm
        cmpq    $1, %rax
        jl      .L102
        ret
.L102:
        movq    $2, %rbx
        subq    %rax, %rbx
        movq    %rbx, %rax
        ret
```

Reply:
I would like to keep same behaviors than OCaml:
```ocaml
# abs min_int
- : int = -4611686018427387904
# Int32.abs Int32.min_int
- : int32 = -2147483648l
```

But I like your trick 👍